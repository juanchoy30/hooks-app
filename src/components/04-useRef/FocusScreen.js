import React, { useRef } from 'react';
import '../02-useEffect/effects.css';

export const FocusScreen = () => {

    const inputRef = useRef();
    // console.log(ref);

    const handClick = () => {
        inputRef.current.select();  // it would be the same to put document.querySelector('input').select();
        // console.log(inputRef);
    }


    return (
        <div>
            <h1>FocusScreen</h1>
            <hr />

            <input 
                ref={ inputRef }
                className="form-control"
                placeholder="Your name"
            />

            <button 
                className="btn btn-outline-primary mt-3"
                onClick={ handClick }
            >
                Focus
            </button>
        </div>
    )
}
