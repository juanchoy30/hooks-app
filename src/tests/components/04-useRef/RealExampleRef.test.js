import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Tests on <RealExampleRef />', () => {

    const wrapper = shallow( <RealExampleRef /> );
    
    test('should render correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );

    });
    
    test('should show the component <MultipleCustomHooks />', () => {
        
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );

    })
    

})
