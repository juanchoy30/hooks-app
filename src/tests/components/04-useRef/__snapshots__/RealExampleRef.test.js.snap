// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <RealExampleRef /> should render correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
